### PR TITLE
Added ANSI support for server GUI

### PR DESCRIPTION
**Pull request by BomBardyGamer**
Likely another useless PR, but I thought I might as well, as I really want to contribute to Paper.
This PR adds support for ANSI colour codes in the text box in the server GUI.

### CODE REVIEW

File: Spigot-Server-Patches/0703-ANSI-support-for-server-GUI.patch [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+Subject: [PATCH] ANSI support for server GUI
+
+
+diff --git a/src/main/java/com/destroystokyo/paper/gui/ANSITextPane.java b/src/main/java/com/destroystokyo/paper/gui/ANSITextPane.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..31405fdc15733c9ab88b2340a268552eac6c3d98
+--- /dev/null
++++ b/src/main/java/com/destroystokyo/paper/gui/ANSITextPane.java
+@@ -0,0 +1,120 @@
++package com.destroystokyo.paper.gui;
```
Comment (kashike) [inline Spigot-Server-Patches/0703-ANSI-support-for-server-GUI.patch]: Wrong package.

File: Spigot-Server-Patches/0703-ANSI-support-for-server-GUI.patch [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
++
++import javax.swing.*;
++import javax.swing.text.*;
++import java.awt.*;
```
Comment (kashike) [inline Spigot-Server-Patches/0703-ANSI-support-for-server-GUI.patch]: No wildcard imports.

File: Spigot-Server-Patches/0703-ANSI-support-for-server-GUI.patch [hunk 3]
```diff
@@ -0,0 +1,272 @@
...
++import com.destroystokyo.paper.gui.ANSITextPane; // Paper - add support for ANSI
+ import com.google.common.collect.Lists;
+ import com.mojang.util.QueueLogAppender;
+ import java.awt.BorderLayout;
+@@ -18,13 +19,14 @@ import javax.swing.JList;
+ import javax.swing.JPanel;
+ import javax.swing.JScrollBar;
+ import javax.swing.JScrollPane;
+-import javax.swing.JTextArea;
++//import javax.swing.JTextArea; // Paper - add support for ANSI
```
Comment (kashike) [inline Spigot-Server-Patches/0703-ANSI-support-for-server-GUI.patch]: Unnecessary change - unused import is fine.

File: Spigot-Server-Patches/0703-ANSI-support-for-server-GUI.patch [hunk 4]
```diff
@@ -0,0 +1,272 @@
...
+ import javax.swing.JTextField;
++import javax.swing.JTextPane; // Paper - add support for ANSI
+ import javax.swing.SwingUtilities;
+ import javax.swing.UIManager;
+ import javax.swing.border.EtchedBorder;
+ import javax.swing.border.TitledBorder;
+-import javax.swing.text.AttributeSet;
++//import javax.swing.text.AttributeSet; // Paper - add support for ANSI
```
Comment (kashike) [inline Spigot-Server-Patches/0703-ANSI-support-for-server-GUI.patch]: Unnecessary change - unused import is fine.

File: Spigot-Server-Patches/0703-ANSI-support-for-server-GUI.patch [hunk 5]
```diff
@@ -0,0 +1,272 @@
...
+ import com.mojang.util.QueueLogAppender;
+ import java.awt.BorderLayout;
+@@ -18,13 +19,14 @@ import javax.swing.JList;
+ import javax.swing.JPanel;
+ import javax.swing.JScrollBar;
+ import javax.swing.JScrollPane;
+-import javax.swing.JTextArea;
++//import javax.swing.JTextArea; // Paper - add support for ANSI
+ import javax.swing.JTextField;
++import javax.swing.JTextPane; // Paper - add support for ANSI
```
Comment (kashike) [inline Spigot-Server-Patches/0703-ANSI-support-for-server-GUI.patch]: Unnecessary change - fully-qualifying is fine.

File: Spigot-Server-Patches/0703-ANSI-support-for-server-GUI.patch [hunk 6]
```diff
@@ -0,0 +1,272 @@
...
++    }
++}
+diff --git a/src/main/java/net/minecraft/server/gui/ServerGUI.java b/src/main/java/net/minecraft/server/gui/ServerGUI.java
+index d1d98c2f192514a3f511bebb00c088b03e6be84c..e47f01c2fa8bdb0468180593db85b7e1faa0d456 100644
+--- a/src/main/java/net/minecraft/server/gui/ServerGUI.java
++++ b/src/main/java/net/minecraft/server/gui/ServerGUI.java
+@@ -1,5 +1,6 @@
+ package net.minecraft.server.gui;
+ 
++import com.destroystokyo.paper.gui.ANSITextPane; // Paper - add support for ANSI
```
Comment (kashike) [inline Spigot-Server-Patches/0703-ANSI-support-for-server-GUI.patch]: Unnecessary change - fully-qualifying is fine.

File: Spigot-Server-Patches/0703-ANSI-support-for-server-GUI.patch [hunk 7]
```diff
@@ -0,0 +1,272 @@
...
++                    insertString(jtextarea.getDocument(), s);
++                }
++            } else {
++                insertString(jtextarea.getDocument(), s);
+             }
++//            try {
++//                document.insertString(document.getLength(), ANSI.matcher(s).replaceAll(""), (AttributeSet) null); // CraftBukkit
++//            } catch (BadLocationException badlocationexception) {
++//                ;
++//            }
```
Comment (kashike) [inline Spigot-Server-Patches/0703-ANSI-support-for-server-GUI.patch]: Use `/*` and `*/` around the block instead of commenting each individually.

**Review by kashike — CHANGES_REQUESTED** [review]
Some stuff I saw, probably more.

File: Spigot-Server-Patches/0703-ANSI-support-for-server-GUI.patch (hunk 1 above)
Comment (BomBardyGamer) [reply Spigot-Server-Patches/0703-ANSI-support-for-server-GUI.patch]: I only put it in there since that's where all the other GUI stuff is. I'm assuming that I should relocate it to `io.papermc.paper.gui`?

Comment (Proximyst) [reply Spigot-Server-Patches/0703-ANSI-support-for-server-GUI.patch]: Yes

File: LICENSE.md [hunk 8]
```diff
@@ -47,4 +47,5 @@ Machine_Maker <machine@machinemaker.me>
 Ivan Pekov <ivan@mrivanplays.com>
 Camotoy <20743703+Camotoy@users.noreply.github.com>
 Bjarne Koll <lynxplay101@gmail.com>
-```
+Callum Seabrook (BomBardyGamer) <callum.seabrook@prevarinite.com>
```
Comment (zachbr) [inline LICENSE.md:50]: This needs to match the email in the patch below

Comment (BomBardyGamer) [reply LICENSE.md:50]: God damn it. Why is my Git email set to `yutubebroz@gmail.com` still, I'm an idiot. Let me try and fix that (if you can even rebase and change the email)

Comment (BomBardyGamer) [reply LICENSE.md:50]: Sorted that, thanks

File: Spigot-Server-Patches/0703-ANSI-support-for-server-GUI.patch (hunk 7 above)
Comment (BomBardyGamer) [reply Spigot-Server-Patches/0703-ANSI-support-for-server-GUI.patch]: Blame IntelliJ for that one

**Comment by electronicboy** [comment]
We're generally not fond of author tags on classes outside of substantial development cases, i.e. you spent all month working on a mega super fast low overhead class which blows the water outside of std-impl, especially as it opens many questions in terms of the `Ship of Theseus`, especially when it honestly looks like most of the impl was a copy paste modification of some code elsewhere and not an original impl, which creates many other caveats around ownership.

In general, I'm also not personally fond of this PR, but, then again, I think that most of the console stuff is generally pointless crud nobody really uses, but, 🤷 
I'd be much happier if this actually did something like offered a terminal which jline supports for aspects like tab completion to work, rather than just a shallow redressing of the current UI

**Comment by BomBardyGamer** [comment]
Alright, I'll see what I can do about the JLine suggestion, and I'll remove my author tags. Thanks.
